tests/Type/data/custom-eloquent-collection.php

```diff
@@ -116,6 +116,9 @@ function test(): void
     assertType('App\TransactionCollection<int, App\Transaction>', (new User)->transactions()->find([1, 2]));
     assertType('Illuminate\Database\Eloquent\Collection<int, App\User>', (new Role)->users()->find([1, 2]));
     assertType('App\Transaction|null', (new User)->transactions()->find(1));
+    assertType('App\Transaction|null', (new User)->transactions()->first());
+    assertType('App\Transaction', (new User)->transactions()->firstOrFail());
+    assertType('App\Transaction|string', (new User)->transactions()->firstOr(callback: fn () => ''));
```

Comment:
You should also have similar tests in one of the `model-relations` files

Reply:
Thanks, I added something similar to the `model-relations` file. Please check if this is what you had in mind.

Reply:
Thanks, now that I think about it these tests can be removed now that you are similar ones to model-relations ince they really have nothing to do with *custom collections*